### PR TITLE
Nitweb: api for code highlighter and nitpicker

### PR DESCRIPTION
**Pull request by privat**
Only the low-level JSON-REST API  is provided. Someone should have to do the rendering.

ping @Morriar

### CODE REVIEW

**Comment by Morriar** [comment]
+1

What kind of frontend do you want?

File: src/examples/nitlight_as_a_service.nit [hunk 1]
```diff
@@ -102,6 +102,24 @@ class HLCode
...
+	# A json content to send as an API
+	fun to_json: Writable
+	do
+		var res = new Template
+		res.add "\{code=\""
+		res.add hl.html.write_to_string.escape_to_c
+		res.add "\",messages=[\n"
+		for m in source.messages do
+			res.add "\{line="
+			res.add((m.location.line_start-1).to_s)
```
Comment (jcbrinfo) [inline src/examples/nitlight_as_a_service.nit]: Why `-1`?

**Comment by privat** [comment]
reroll this old PR. try to pass tests

**Comment by privat** [comment]
test this please

File: src/highlight.nit [hunk 2]
```diff
@@ -20,6 +20,105 @@ import html
...
+		end
+		res.add """});}"""
+		return res
+	end
+
+	# A json content to send as an API
+	fun to_json: Writable
+	do
+		var res = new Template
+		res.add "\{code=\""
```
Comment (jcbrinfo) [inline src/highlight.nit]: The thing is that is not JSON.

Comment (privat) [inline src/highlight.nit]: I see your point

**Comment by privat** [comment]
test this please

**Pull request merged**